model/postgres/postgres_timeline.rb

```diff
@@ -58,7 +58,8 @@ def backups
 
   def latest_backup_label_before_target(target:)
     backup = backups.sort_by(&:last_modified).reverse.find { _1.last_modified < target }
-    backup.key.delete_prefix("basebackups_005/").delete_suffix("_backup_stop_sentinel.json") if backup
+    fail "BUG: no backup found" unless backup
```

Comment:
is there a reason this wasn't a bug before, but is a bug now?

Reply:
It was a bug, but the check was in the caller's code (i.e. PostgresServer). I moved it here because I though it is the timeline's responsibility not the callers.